lib/views/pages/profile_page.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_ecommerce/controllers/auth_controller.dart';
+import 'package:flutter_ecommerce/views/pages/auth_page.dart';
+import 'package:flutter_ecommerce/views/widgets/main_button.dart';
+import 'package:provider/provider.dart';
+import '../../services/auth.dart';
+import '../../utilities/enums.dart';
+import '../../utilities/routes.dart';
+import '../widgets/default_toast.dart';
+
+class ProfilePage extends StatelessWidget {
+  const ProfilePage({Key? key}) : super(key: key);
+
+
+  Future<void> _logout(AuthController model, context) async {
+    try {
+      await model.logout();
+      const DefaultToast(message: 'successfully logged out', state: ToastStates.success)
+          .showToast();
+      Navigator.pushAndRemoveUntil(context,  MaterialPageRoute(
```

Comment:
You can just pop after logging out because you mean you will go to the landing page and it's automatically will redirect you to the auth page